src/reducers/entities/posts.js

```diff
@@ -39,7 +39,10 @@ function handleReceivedPost(posts = {}, postsInChannel = {}, postsInThread = {},
         ];
     }
 
-    return {posts: nextPosts, postsInChannel: nextPostsInChannel, postsInThread: nextPostsInThread};
+    const withCombineSystemPosts = combineSystemPosts(nextPostsInChannel[channelId], nextPosts);
+    nextPostsInChannel[channelId] = withCombineSystemPosts.postsForChannel;
+
+    return {posts: withCombineSystemPosts.nextPosts, postsInChannel: nextPostsInChannel, postsInThread: nextPostsInThread};
```

Comment:
`nextPostsInChannel` and `nextPostsInThread` here should have the ids for those posts that were removed from the combine stuff as you will end up having something like this for example: for 3 system messages in a channel you'll have 1 post in the posts entity and 3 posts in the postsInChannel entity.

sounds about right? probably the cause that you needed to filter later on

Reply:
Got it, I'll update this.

Reply:
Can confirm that those system posts are included in `nextPostsInChannel` but not necessary in `nextPostsInThread`. Those system posts on `nextPostsInChannel` were marked as deleted.